strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
Well in this case yes - we assume here that we always deal with reports in JUnit format because we filter for it using predicate. This is not pluggable and is equal to `new JUnitParser`.